sdk/core/allocator/alloc.h

```diff
@@ -2811,4 +2811,163 @@ class MState
 	{
 		ABORT();
 	}
+
+#if HEAP_RENDER
+	public:
+	/**
+	 * "Render" the heap for debugging.
+	 */
+	template<bool Asserts = false, bool Chatty = true>
```

Comment:
I think Asserts should default to `true`. They don't look that expensive compared to the debug logs (I guess `ds::linked_list::is_well_formed` may be linear in length of list?).

Reply:
Thanks!

Reply:
> I guess `ds::linked_list::is_well_formed` may be linear in length of list?

No, is just a check of the well-formedness of the node.  Could probably be better named, in retrospect.